### PR TITLE
ISSUE-211 Implement a LongLivedToken authentication strategy

### PR DESCRIPTION
**Pull request by vttranlina**
ref https://github.com/linagora/tmail-backend/pull/225

### CODE REVIEW

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/longlivedtoken/LongLivedTokenAuthenticationStrategy.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  override def createMailboxSession(httpRequest: HttpServerRequest): Mono[MailboxSession] =
+    SMono.justOrEmpty(retrieveAuthenticationToken(httpRequest))
+      .flatMap(token => SMono.fromPublisher(longLivedTokenStore.validate(token.username, token.secret))
+        .map(_ => sessionProvider.createSystemSession(token.username))
+        .switchIfEmpty(SMono.error(new UnauthorizedException("Invalid long lived token"))))
+      .asJava()
+
+  override def correspondingChallenge(): AuthenticationChallenge = AuthenticationChallenge.of(
+    AuthenticationScheme.of("Ticket"),
```
Comment (Arsnael) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/longlivedtoken/LongLivedTokenAuthenticationStrategy.scala:33]: Nope...
Should be `Bearer`

File: tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/longlivedtoken/LongLivedTokenAuthenticationStrategyTest.scala [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      .thenReturn("Bearer " + TOKEN)
+    when(longLivedTokenStore.validate(ArgumentMatchers.eq(Username.of(USER_NAME)), ArgumentMatchers.eq(SECRET_UUID)))
+      .thenReturn(SMono.empty)
+
+    assertThatCode(() => testee.createMailboxSession(mockedRequest).block())
+      .isInstanceOf(classOf[UnauthorizedException])
+  }
+
+  @Test
+  def createMailboxSessionShouldReturnWhenAuthHeadersAreValid(): Unit = {
```
Comment (Arsnael) [inline tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/longlivedtoken/LongLivedTokenAuthenticationStrategyTest.scala:82]: `createMailboxSessionShouldReturnValidSessionWhenAuthHeadersAreValid`

File: tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/longlivedtoken/LongLivedTokenAuthenticationStrategyTest.scala [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+  @ParameterizedTest
+  @ValueSource(strings = Array(
+    "bob_bob@linagora",
+    "_bob",
+    "_bob_",
+    "_bob__",
+    "bob__@a"
+  ))
+  def createMailboxSessionShouldReturnWhenAuthHeaderHasSeveralSpecialCharacter(username: String): Unit = {
```
Comment (Arsnael) [inline tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/longlivedtoken/LongLivedTokenAuthenticationStrategyTest.scala:103]: `createMailboxSessionShouldReturnValidSessionWhenAuthHeaderHasSeveralSpecialCharacter`

**Review by Arsnael — COMMENTED** [review]
Seems good overall, just few comments :)

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**